docs/admin/general-upgrade-guidelines.rst

```diff
@@ -0,0 +1,73 @@
+.. highlight:: sh
+
+.. _general_upgrade_guidelines:
+
+==========================
+General upgrade guidelines
+==========================
+
+Before kicking off an upgrade, there is a set of best practices to ensure the best outcome. Below you may find the five fundamental steps to prepare for an upgrade.
+
+.. NOTE::
+
+   This is not an exhaustive list, so you should consider your organization's specific needs and incorporate any additional steps or considerations that are relevant to your environment.
+
+Upgrade planning
+================
+
+Acknowledge breaking changes
+-----------------------------
+
+Review the `release notes`_ and documentation for the target version to understand any potential impact on existing functionality. 
+In case of a minor version upgrade, you should consider the intermediate versions. For example, when upgrading from 5.1 to 5.3, besides reviewing 5.3 release notes, the recommendation is to check for version 5.2 as well.
+
+Set up a test environment
+-------------------------
+
+Create a test environment that closely resembles your production environment, including the same CrateDB version, hardware, and network configuration. Populate the test environment with representative data and perform thorough testing to ensure compatibility and functionality, including functional and non-functional testing.
+
+
+Back up and plan recovery
+-------------------------
+
+Perform a cluster-wide backup of your production CrateDB and ensure you have a reliable recovery mechanism in place. Read more in the `snapshots`_ documentation. Once the backup is complete, validate it to ensure its usability.
+
+For the newly written records, you should consider using a mechanism to queue them (e.g. message queue), so these messages can be replayed if needed.
+
+.. WARNING::
+   
+   Before starting the upgrade process, ensure no backup processes are triggered, so disable any scheduled backup.
+
+Define a rollback plan
+-----------------------
+
+The rollback plan may vary depending on the specific infrastructure and upgrade process in use. It is also essential to adapt this outline to your organization's specific needs and incorporate any additional steps or considerations that are relevant to your environment. A set of steps to serve as an example is listed below:
```

Comment:
One additional step would also be to identify the rollback process, depending on the version jump. It doesn't necessarily need to be a backup restore. E.g. if upgrading from one patch release to another, and there is no data corruption but only a performance issue, a simple in-place downgrade to the previous patch release is sufficient.
Restoring from a backup would only be needed for major/minor version jumps or in case of data corruption.